.github/workflows/build.yaml

```diff
@@ -0,0 +1,100 @@
+name: Build
+
+on:
+  pull_request:
+
+permissions:
+  contents: read
+
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+jobs:
+  ci:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: windows-2022
+            uploaddiff: 1
+            preset: windows-ci
+          - os: ubuntu-20.04
+            preset: linux-ci
+          - os: macos-12
+            preset: macos-ci
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
+    env:
+      VCPKG_ROOT: ${{ github.workspace }}/vcpkg-root
+      DiffFile: ${{ github.workspace }}/out/format.diff
+    steps:
+      - uses: actions/checkout@v3
+      - name: 'Format C++'
+        if: matrix.uploaddiff
+        shell: pwsh
+        run: ./azure-pipelines/Format-CxxCode.ps1
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "18.x"
+          cache: 'npm'
+          cache-dependency-path: vcpkg-artifacts/package-lock.json
+      - uses: lukka/get-cmake@v3.26.3
+        with:
+          cmakeVersion: 3.22.1 # ubuntu 22.04
+          ninjaVersion: 1.11.1
+      - name: Configure and Build
+        if: matrix.os != 'windows-2022'
+        run: |
+          cmake --preset ${{ matrix.preset }}
+          cmake --build --preset ${{ matrix.preset }} -- -k0
+      - name: Configure and Build (Windows)
+        if: matrix.os == 'windows-2022'
+        shell: cmd
+        run: |
+          call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\Tools\vsdevcmd.bat" -arch=amd64 -host_arch=amd64
+          IF %ERRORLEVEL% NEQ 0 exit /B %ERRORLEVEL%
+          cl /Bv
+          set CXX=cl
+          set CC=cl
+          cmake --preset ${{ matrix.preset }}
+          IF %ERRORLEVEL% NEQ 0 exit /B %ERRORLEVEL%
+          cmake --build --preset ${{ matrix.preset }} -- -k0
+          IF %ERRORLEVEL% NEQ 0 exit /B %ERRORLEVEL%
+      - name: Regenerate Message Map
+        if: matrix.uploaddiff
+        run: cmake --build --preset ${{ matrix.preset }} --target generate-message-map
+      - name: Create Diff
+        if: matrix.uploaddiff
+        shell: pwsh
+        run: azure-pipelines/Create-PRDiff.ps1 -DiffFile $env:DiffFile
+      - name: Publish Format and Messages File Diff
+        uses: actions/upload-artifact@v3
+        if: matrix.uploaddiff && failure()
+        with:
+          name: 'format.diff'
+          path: ${{ env.DiffFile }}
+      - name: Run vcpkg unit tests
+        run: ctest --preset ${{ matrix.preset }}
+        env:
+          VCPKG_ROOT: UNIT_TESTS_SHOULD_NOT_USE_VCPKG_ROOT
+      - name: Run vcpkg-artifacts unit tests
+        run: |
+          cd out/build/${{ matrix.preset }}/vcpkg-artifacts
+          node node_modules/mocha/bin/mocha --config mocha-config.yaml
```

Comment:
No change requested for now: should we hook mocha up as a test so you don't need the second step here?

Reply:
I do think that would be good, but I wanted to do that as a separate PR after pipelines.yml is removed (so we don't have to make the change in two places)